datafusion/core/src/datasource/listing/mod.rs

```diff
@@ -58,6 +59,8 @@ pub struct PartitionedFile {
     pub partition_values: Vec<ScalarValue>,
     /// An optional file range for a more fine-grained parallel execution
     pub range: Option<FileRange>,
+    /// An optional field for user defined per object metadata  
+    pub metadata_ext: Option<FileMetaExt>,
```

Comment:
I wonder if `Arc<dyn std::any::Any>` might be better? And just call it `extensions`?